ArjunaJTA/jdbc/classes/com/arjuna/ats/internal/jdbc/ConnectionImple.java

```diff
@@ -1002,7 +1003,7 @@ protected final synchronized void registerDatabase() throws SQLException
 							jdbcLogger.i18NLogger.debug_closingconnection(_theConnection.toString());
 
 							delayClose = true;
-							tx.registerSynchronization(new ConnectionSynchronization(this, needsClose));
+							jtaPropertyManager.getJTAEnvironmentBean().getTransactionSynchronizationRegistry().registerInterposedSynchronization(new ConnectionSynchronization(this, needsClose));
```

Comment:
It fixes JBTM-2909 only. It must be interposed so HHH can do its work in the AC also.